b2share/modules/deposit/api.py

```diff
@@ -112,6 +128,36 @@ def create(cls, data, id_=None):
         # Set record's schema
         if '$schema' in data:
             raise InvalidDepositError('"$schema" field should not be set.')
+
+        if '_parent_pids' in data:
+            raise InvalidDepositError(
+                'Field "_parent_pids" cannot be set.')
+        version_master, prev_version = None, None
+        if version_of:
+            version_master, prev_version = find_version_master_and_previous_record(version_of)
+
+        if prev_version:
+            if not get_version_parent_pid(prev_version):
+                # previous version needs updating with the versioning parent pid
+                set_version_parent_pid(prev_version, version_master.parent)
+                prev_version.commit()
+                RecordIndexer().index(prev_version)
+
+            if 'community' in data and data['community'] != prev_version.model.json['community']:
+                raise InvalidDepositError(
+                    'Community ID must be identical with '
+                    'the previous version\'s community ID.')
+            if current_user.id not in prev_version['_deposit']['owners']:
+                raise OwnershipVersioningError()
+            patch = data
+            data = copy.deepcopy(prev_version.model.json)
+            # eliminate _deposit, _files, _oai, _pid, etc.
+            data = {k: v for k, v in data.items() if not k.startswith('_')}
+            data.update(patch)
```

Comment:
Do we really want to keep by default the values of the previous record? This will create problems with records created via the REST API.

Reply:
This is a very important feature for usability, we should do it even if it costs us. But I don't yet see what problems it can create, can you clarify?

Reply:
If you create a record via REST API I expect that you already have all the data. Ex: LTER records. There might be unexpected side effects if a field is renamed in the community schema. The field with the old name would be copied without the client noticing.
For UI at least the user checks the content before publishing. Then copying makes more sense IMHO. Although there would still be the issue of schema change.

Reply:
Yes you're right, this should not be handled here in the programmatic api but in the UI, I'll make the changes.